src/MOIWrapper.jl

```diff
@@ -0,0 +1,345 @@
+export CbcOptimizer
+
+
+using MathOptInterface
+using Cbc.CbcCInterface
+
+const MOI = MathOptInterface
+const MOIU = MathOptInterface.Utilities
+const CbcCI = CbcCInterface
+
+mutable struct CbcOptimizer <: MOI.AbstractOptimizer
+    inner::CbcModel
+    CbcOptimizer() = new(CbcModel()) # Initializes with an empty model
+end
+
+struct CbcModelFormat
+    nbRows::Int
+    nbCols::Int
+    constraint_matrix::SparseMatrixCSC{Float64,Int}
+    col_lb::Vector{Float64}
+    col_ub::Vector{Float64}
+    obj::Vector{Float64}
+    row_lb::Vector{Float64}
+    row_ub::Vector{Float64}
+    function CbcModelFormat(nbRows::Int, nbCols::Int)
+        obj = fill(0.0, nbCols)
+        col_lb = fill(-Inf, nbCols)
+        col_ub = fill(Inf, nbCols)
+        row_lb = fill(-Inf, nbRows)
+        row_ub = fill(Inf, nbRows)
+        constraint_matrix = sparse(Int[], Int[], Float64[], nbRows, nbCols)
+        new(nbRows, nbCols, constraint_matrix, col_lb, col_ub, obj, row_lb, row_ub)
+    end
+end
+
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.SingleVariable, s::MOI.EqualTo)
+    cbcModelFormat.col_lb[mapping.varmap[f.variable].value] = s.value
+    cbcModelFormat.col_ub[mapping.varmap[f.variable].value] = s.value
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.SingleVariable, s::MOI.LessThan)
+    cbcModelFormat.col_ub[mapping.varmap[f.variable].value] = s.upper
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.SingleVariable, s::MOI.GreaterThan)
+    cbcModelFormat.col_lb[mapping.varmap[f.variable].value] = s.lower
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.SingleVariable, s::MOI.Interval)
+    cbcModelFormat.col_lb[mapping.varmap[f.variable].value] = s.lower
+    cbcModelFormat.col_ub[mapping.varmap[f.variable].value] = s.upper
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat,
+    mapping::MOIU.IndexMap, f::MOI.ScalarAffineFunction, s::MOI.EqualTo)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,
+        mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_lb[mapping.conmap[ci].value] = s.value - f.constant
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.value - f.constant
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat,
+    mapping::MOIU.IndexMap, f::MOI.ScalarAffineFunction, s::MOI.GreaterThan)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,
+        mapping.varmap[term.variable_index].value] += term.coefficient
```

Comment:
This is a super inefficient way to build a sparse matrix. Instead, keep a collection of the `(row, column, value)` triplets and call `sparse()` to build the matrix at the end.